### PR TITLE
Coming Soon v2: roll back page selector

### PR DESCRIPTION
**Pull request by ramonjd**
#### Changes proposed in this Pull Request

This commit reverts all of the changes from #45892, which adds a Coming Soon page selector to the page list.

We don't need the page selector feature right now but might need them later, in which case we'll re-add them.

We've also deleted the unused selectors. `isComingSoonModeActive` might be useful in the future but, for now we're not using it.

We're keeping the reducer changes since they effect the settings page cc @lsl 

#### Testing instructions

Check the testing instructions in https://github.com/Automattic/wp-calypso/pull/45892

The Coming Soon option should not be present in the page selector.

Tracking should also be gone.

### CODE REVIEW

File: client/my-sites/pages/page/index.js [hunk 1]
```diff
@@ -354,54 +352,6 @@ class Page extends Component {
...
-	setComingSoonPage = () => {
-		const pageId = this.props.isComingSoonPage ? 0 : this.props.page.ID;
-
-		this.props.updateSiteFrontPage( this.props.siteId, {
-			wpcom_public_coming_soon_page_id: pageId,
-		} );
-
-		// records whether a page has been set as 'coming soon'
-		// and the paid plan slug
-		this.props.recordTracksEvent( 'calypso_coming_soon_set_page', {
```
Comment (p-jackson) [inline client/my-sites/pages/page/index.js:366]: Did this event ever get registered? Is there such a thing as unregistering an event?

Comment (p-jackson) [reply client/my-sites/pages/page/index.js:366]: I don't see it in the repo, so no it wasn't registered 👍

**Review by p-jackson — COMMENTED** [review]
I think we _do_ want to remove some of the other changes to:
- `client/state/data-layer/wpcom/sites/homepage/index.js` It includes code for setting the `wpcom_public_coming_soon_page_id` site option, which we're no longer going to have.
- `client/state/sites/selectors/get-coming-soon-page-id.js` We don't need this whole selector right? Since the concept of `wpcom_public_coming_soon_page_id` is disappearing.
- `client/state/pages/selectors.js`

File: client/my-sites/pages/page/index.js (hunk 1 above)
Comment (ramonjd) [reply client/my-sites/pages/page/index.js:366]: > wpcom_public_coming_soon_page_id

Ah yeah, fair point. Thanks for highlighting it :) I'll remove

**Comment by ramonjd** [comment]
I rolled everything back except the [site settings reducer change](https://github.com/Automattic/wp-calypso/blob/765f3bb5b5a9c94f92115c9b0623579239247a90/client/state/sites/reducer.js)

**Review by p-jackson — APPROVED** [review]

**Pull request merged**